packages/suite/src/views/dashboard/components/AssetsCard/index.tsx

```diff
@@ -117,33 +136,74 @@ const AssetsCard = () => {
                         );
 
                         return (
-                            <Asset
+                            <AssetGrid
                                 key={symbol}
                                 network={network}
                                 failed={!!assetFailed}
                                 cryptoValue={assetBalance.toFixed()}
-                                isLastRow={i === networks.length - 1}
                             />
                         );
                     })}
-                </Grid>
-                {isLoading && (
-                    <InfoMessage>
-                        <Loader size={20} />
-                    </InfoMessage>
-                )}
-                {isError && (
-                    <InfoMessage>
-                        <Icon
-                            style={{ paddingRight: '4px', paddingBottom: '2px' }}
-                            icon="WARNING"
-                            color={theme.TYPE_RED}
-                            size={14}
-                        />
-                        <Translation id="TR_DASHBOARD_ASSETS_ERROR" />
-                    </InfoMessage>
-                )}
-            </StyledCard>
+                </GridWrapper>
+            )}
+            {isTableMode && (
+                <StyledCard>
+                    <Grid>
+                        <Header>
+                            <Translation id="TR_ASSETS" />
+                        </Header>
+                        <Header>
+                            <Translation id="TR_VALUES" />
+                        </Header>
+                        <Header>
+                            <Translation id="TR_EXCHANGE_RATE" />
+                        </Header>
+                        {networks.map((symbol, i) => {
+                            const network = NETWORKS.find(
+                                n => n.symbol === symbol && !n.accountType,
+                            );
+                            if (!network) {
+                                return 'unknown network';
+                            }
+
+                            const assetBalance = assets[symbol].reduce(
+                                (prev, a) => prev.plus(a.formattedBalance),
+                                new BigNumber(0),
+                            );
+
+                            const assetFailed = accounts.find(
+                                f => f.symbol === network.symbol && f.failed,
+                            );
```

Comment:
This is redundant with L124-136. Data could be prepared and memoized higher up before being mapped. The 'unknown error' message is not very relevant and I guess should be logged as an error instead.

Reply:
Done